pandas/_libs/tslib.pyx

```diff
@@ -561,6 +600,24 @@ cpdef array_to_datetime(
         else:
             tz_offset = out_tzoffset_vals.pop()
             tz_out = timezone(timedelta(seconds=tz_offset))
+
+    if infer_reso:
+        if state.creso_ever_changed:
```

Comment:
Not sure what to do here but my first read of `state.creso_ever_changed` was confusion over the scope of `ever`; I am inferring from the way the function is currently written that `ever` refers to the lifetime of this function, but if we have parse states that can persist across multiple function calls that terminology can get a little vague

Reply:
> if we have parse states that can persist across multiple function calls

we do not

Reply:
Any reason to keep it as part of the state then instead of just local to the function?

To be clear not a hold up on this PR for me. Just a consideration point as this continues to evolve

Reply:
bc it gets set inside a state-updating method that i dont want to duplicate in a bunch of places.  also there are a couple of different places where we use DatetimeParseState and im trying to iron out the kinks in behavior differences between them.